src/components/cluster/detail/ScaleClusterModal.js

```diff
@@ -331,6 +331,9 @@ class ScaleClusterModal extends React.Component {
               this.props.provider,
               this.props.cluster.release_version
             )}
+            valueConstraints={{
+              max: this.props.maxWorkersPerClusters,
+            }}
             onChange={this.updateScaling}
```

Comment:
I think naming could be improved here. On AWS, cluster sizes are configured by the users in a range between `min` and `max`. On Azure and KVM, for now, instead there is only one value for the desired number of worker nodes.

If we set constraints for these, we should make sure that it is clear to what they apply. For example, there could be a lower and an upper bound for the `min` value and likewise for the `max` value. Or there could be an absolute minimum and maximum of worker nodes to permit.

If necessary, please clarify in comments what this constraint really applies to.

Reply:
For AWS, the value constraints here (minimum and maximum) apply to the overall values. (the minimum constraint value applies as a minimum value for the `MIN` number selector, and the maximum constraint value applies as a maximum value for the `MAX` number selector.

For KVM and Azure, they act as constraints for the single number selector.

As for clarification, there is no need to set individual limits for each number selector.
With this, the `MIN` number selector has the minimum value of the `min` constraint, and the maximum value of `MAX`'s current value; the `MAX` number selector has a minimum value of `MIN`'s current value and the maximum value of the `min constraint`

Let me know if you need any more information

Reply:
All good!